osprey_flask_app/routes.py

```diff
@@ -44,17 +46,51 @@ def osprey_route():
     """
     args = request.args
     arg_dict = create_full_arg_dict(args)
-    outpath = run_full_rvic(arg_dict)
+    validate_response = validate_inputs(arg_dict)
+    if validate_response.status_code == 400:
+        return validate_response
+
+    rvic_thread = threading.Thread(
+        target=lambda q, arg: q.put(run_full_rvic(arg)), args=(que, arg_dict)
+    )
+    rvic_thread.start()
+    return Response(
+        "RVIC Process started. Check status: "
+        + url_for("osprey.status_route", thread_id=rvic_thread.native_id),
+        status=202,
+    )
+
+
+@osprey.route("/status/<thread_id>", methods=["GET"])
+def status_route(thread_id):
+    """Provide route to check status of RVIC process."""
+    active_thread_ids = [str(t.native_id) for t in threading.enumerate()]
+    if thread_id in active_thread_ids:
+        return Response("Process is still running.", status=201)
+    else:
+        return Response(
+            "Process completed. Get output: "
+            + url_for("osprey.output_route", thread_id=thread_id),
+            status=200,
+        )
+
+
+@osprey.route("/output/<thread_id>", methods=["GET"])
+def output_route(thread_id):
+    """Provide route to get streamflow output of RVIC process."""
+    if que.empty():
+        return Response("Process has failed. No output returned.", status=404)
     try:
-        outpath_url = requests.get(outpath)
+        outpath = que.get()
+        outpath_response = requests.get(outpath)
     except requests.exceptions.ConnectionError as e:
-        raise
+        return Response("Process has failed. " + e, status=404)
 
     with NamedTemporaryFile(suffix=".nc", dir="/tmp") as outfile:
-        outfile.write(outpath_url.content)
+        outfile.write(outpath_response.content)
```

Comment:
So a) you don't want to use `/tmp` here. You should *at least* use the default temporary directory (which can be set at configuration time by [setting one of the environment variables `TMPDIR, TEMP or TMP`](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp)).

b) if `outpath` here is a URL, instead of reading it and the writing it back out (extra IO!), why not just send a redirect response (status 302) directly to the resource?